x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/ComputeService.java

```diff
@@ -175,8 +175,8 @@ public void execute(
             .groupIndices(SearchRequest.DEFAULT_INDICES_OPTIONS, PlannerUtils.planOriginalIndices(physicalPlan));
         var localOriginalIndices = clusterToOriginalIndices.remove(RemoteClusterAware.LOCAL_CLUSTER_GROUP_KEY);
         var localConcreteIndices = clusterToConcreteIndices.remove(RemoteClusterAware.LOCAL_CLUSTER_GROUP_KEY);
-        if (clusterToOriginalIndices.isEmpty() == false && PlannerUtils.hasEnrich(physicalPlan)) {
-            listener.onFailure(new IllegalArgumentException("cross clusters query doesn't support enrich yet"));
+        if (PlannerUtils.hasUnsupportedEnrich(physicalPlan)) {
+            listener.onFailure(new IllegalArgumentException("Enrich modes COORDINATOR and REMOTE are not supported yet"));
```

Comment:
This is better handled in the Verifier - no need to hold the PR for it though.

Reply:
This will be removed shortly.